### PR TITLE
feat: redeem cycles faucet coupon to cycles ledger

### PR DESCRIPTION
**Pull request by smallstepman**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. 

Fixes # (issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

# Checklist:

- [x] The title of this PR complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] I have edited the CHANGELOG accordingly.
- [ ] I have made corresponding changes to the documentation.

### CODE REVIEW

File: src/dfx/src/commands/cycles/redeem_faucet_coupon.rs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        .get_selected_identity_principal()
+        .with_context(|| anyhow!("No identity selected."))?;
+    let response = agent
+        .update(&faucet_principal, "redeem_to_cycles_ledger")
+        .with_arg(
+            encode_args((
+                opts.coupon_code.clone(),
+                Account {
+                    owner: identity,
+                    subaccount: None,
```
Comment (sesi200) [inline src/dfx/src/commands/cycles/redeem_faucet_coupon.rs:53]: Would be nice to have `--to-subaccount` as an option

File: e2e/tests-dfx/cycles-ledger.bash [hunk 2]
```diff
@@ -544,4 +544,18 @@ current_time_nanoseconds() {
...
+  dfx_new hello
+  install_asset faucet
+  dfx deploy
+  dfx ledger fabricate-cycles --canister faucet --t 1000
+
+  dfx identity new --storage-mode plaintext no_wallet_identity
+  dfx identity use no_wallet_identity
+
+  assert_command dfx cycles redeem-faucet-coupon --faucet "$(dfx canister id faucet)" 'valid-coupon'
+  assert_match "Redeemed coupon 'valid-coupon' to the cycles ledger, current balance: .* TC .* for identity '$(dfx identity get-principal)'"
```
Comment (sesi200) [inline e2e/tests-dfx/cycles-ledger.bash:560]: I'd expect the identity name here if you say `balance for identity <...>`

**Comment by sa-github-api** [comment]
Dear @smallstepman,

In order to potentially merge your code in this open-source repository and therefore proceed with your contribution, we need to have your approval on DFINITY's [CLA](https://github.com/dfinity/cla/blob/main/CLA.md).

If you decide to agree with it, please visit this [issue](https://github.com/dfinity/cla/issues/442) and read the instructions there. Once you have signed it, re-trigger the workflow on this PR to see if your code can be merged.

— The DFINITY Foundation

**Comment by sesi200** [comment]
Closing in favor of #3569 because of CLA

**Pull request closed without merging**